src/Cursor.js

```diff
@@ -37,8 +37,10 @@ function update(cmp, path, operation, nextValue) {
   var nextState;
 
   if (path.length > 0) {
+    var q = cmp._reactInternalInstance._pendingStateQueue;
```

Comment:
I like this, it's clever. I also agree with you that it's better to peek at the last item rather than pop it off the queue.